### PR TITLE
Support musicbrainz fields in search and distinct

### PR DESCRIPTION
**Pull request by kingosticks**
A fix for #42. My first attempt at using SQLite, hope I did the migration correctly.

This requires https://github.com/mopidy/mopidy/pull/1900

### CODE REVIEW

**Comment by tkem** [comment]
@kingosticks: I'm pretty impressed you could actually make some sense out of this mess ;-)
However, I think you should also update `sql/schema.sql`, too, for fresh installs.
AFAICS, only `sql/upgrade-v6.sql` has been changed. So with a fresh install, the v5 schema will be created with `schema.sql`, and then the `upgrade-v6.sql` script will be run. I always thought it would be more clear if `schema.sql` contains the current schema. Just easier by looking at a single file to see what's going on. 
But it's debatable, since that means some code duplication in the upgrade scripts.

File: mopidy_local/sql/upgrade-v6.sql [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+       album_name                       AS album,
+       artist_name                      AS artist,
+       composer_name                    AS composer,
+       performer_name                   AS performer,
+       albumartist_name                 AS albumartist,
+       genre                            AS genre,
+       track_no                         AS track_no,
+       coalesce(date, album_date)       AS date,
+       comment                          AS comment,
+       disc_no                          AS disc_no,
```
Comment (tkem) [inline mopidy_local/sql/upgrade-v6.sql:25]: Move `disc_no` up to `track_no` for aesthetic reasons and to match `_SEARCH_FIELDS` in `schema.py`.

File: mopidy_local/sql/upgrade-v6.sql [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+    album,
+    artist,
+    composer,
+    performer,
+    albumartist,
+    genre,
+    track_no,
+    date,
+    comment,
+    disc_no,
```
Comment (tkem) [inline mopidy_local/sql/upgrade-v6.sql:46]: Move `disc_no` up to `track_no` for aesthetic reasons and to match `_SEARCH_FIELDS` in `schema.py`.

File: mopidy_local/sql/upgrade-v6.sql [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+        album,
+        artist,
+        composer,
+        performer,
+        albumartist,
+        genre,
+        track_no,
+        date,
+        comment,
+        disc_no,
```
Comment (tkem) [inline mopidy_local/sql/upgrade-v6.sql:70]: Move disc_no up to track_no for aesthetic reasons and to match _SEARCH_FIELDS in schema.py.

File: mopidy_local/sql/upgrade-v6.sql [hunk 4]
```diff
@@ -0,0 +1,122 @@
...
```
Comment (tkem) [inline mopidy_local/sql/upgrade-v6.sql:92]: Move disc_no up to track_no for aesthetic reasons and to match _SEARCH_FIELDS in schema.py.

File: mopidy_local/sql/upgrade-v6.sql [hunk 5]
```diff
@@ -0,0 +1,122 @@
...
+    album,
+    artist,
+    composer,
+    performer,
+    albumartist,
+    genre,
+    track_no,
+    date,
+    comment,
+    disc_no,
```
Comment (tkem) [inline mopidy_local/sql/upgrade-v6.sql:114]: Move disc_no up to track_no for aesthetic reasons and to match _SEARCH_FIELDS in schema.py.

**Comment by tkem** [comment]
BTW, since v5 was the first sqlite schema version used in `mopidy-local`, the older upgrade scripts could probably be dropped.

**Comment by kingosticks** [comment]
Ahh, I didn't notice that `sql/schema.sql` actually matched the latest schema (the version number should have been a giveaway). That all makes sense. I've also made those suggested changes.

**Pull request merged**